app/src/main/java/ltd/evilcorp/atox/activity/SettingsActivity.kt

```diff
@@ -0,0 +1,122 @@
+package ltd.evilcorp.atox.activity
+
+import android.content.SharedPreferences
+import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.PreferenceManager
+import ltd.evilcorp.atox.App
+import ltd.evilcorp.atox.R
+import ltd.evilcorp.atox.tox.ToxThread
+
+private const val TITLE_TAG = "settingsActivityTitle"
+
+class SettingsActivity : AppCompatActivity(),
+    PreferenceFragmentCompat.OnPreferenceStartFragmentCallback,
+    SharedPreferences.OnSharedPreferenceChangeListener {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.settings_activity)
+        if (savedInstanceState == null) {
+            supportFragmentManager
+                .beginTransaction()
+                .replace(R.id.settings, HeaderFragment())
+                .commit()
+        } else {
+            title = savedInstanceState.getCharSequence(TITLE_TAG)
+        }
+        supportFragmentManager.addOnBackStackChangedListener {
+            if (supportFragmentManager.backStackEntryCount == 0) {
+                setTitle(R.string.title_activity_settings)
+            }
+        }
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onSharedPreferenceChanged(prefs: SharedPreferences?, key: String?) {
+        Log.e("Settings", "Key changed $key")
+        when {
+            key.equals("username") -> {
+                with(App.toxThread.handler) {
+                    sendMessage(obtainMessage(ToxThread.msgSetName, 0, 0, prefs!!.getString(key, "aTox User")))
```

Comment:
please make the default text unique for all aTox users.
like "aTox User F3A04FF1" or something
maybe use the last part of the Tox Pubkey?
otherwise on someone elses client they wil get 10 users all called "aTox User"
:-(

Reply:
Maybe eventually, but not in this PR. I'm not even sure we'll leave the username part optional when opening the client the first time.